### PR TITLE
[ARGG-794][BpkDataTable] Remove deprecated props in BpkDataTable and migrate to TS

### PR DESCRIPTION
**Pull request by anambl**
Migration guide:
https://github.com/Skyscanner/backpack/blob/main/packages/bpk-component-datatable/docs/migrating.md

Remember to include the following changes:

- [x] Ensure the PR title includes the name of the component you are changing so it's clear in the release notes for consumers of the changes in the version e.g `[KOA-123][BpkButton] Updating the colour`
- [ ] `README.md` (If you have created a new component)
- [ ] Component `README.md`
- [x] Tests
- [x] Storybook examples created/updated
- [x] For breaking changes or deprecating components/properties, migration guides added to the description of the PR. If the guide has large changes, consider creating a new Markdown page inside the component's docs folder and link it here

### CODE REVIEW

File: examples/bpk-component-datatable/examples.tsx [hunk 1]
```diff
@@ -74,8 +73,8 @@ const complexRows = [
 const onRowClick = (row) => alert(JSON.stringify(row));
 
 const sortFunc = (rowA, rowB) => {
-  const deskA = rowA.values.seat.desk;
-  const deskB = rowB.values.seat.desk;
+  const deskA = rowA.seat.desk;
```
Comment (anambl) [inline examples/bpk-component-datatable/examples.tsx:76]: This was a bit of a bug as the row object was the one coming from `react-table` which had a lot of unnecessary properties on it. I've created a function to abstract away the `react-table` API and the row will be just the object that the consumer passes

File: packages/bpk-component-datatable/src/BpkDataTable.tsx [hunk 2]
```diff
@@ -122,17 +116,17 @@ const BpkDataTable = (props: Props) => {
     );
 
   const onRowClicked = (index: number) => {
-    if (rowSelected === index) {
-      updateRowSelected(undefined);
-    } else {
-      updateRowSelected(index);
-    }
     if (onRowClick !== undefined) {
+      if (rowSelected === index) {
```
Comment (anambl) [inline packages/bpk-component-datatable/src/BpkDataTable.tsx:120]: moved this inside the `if` as previously it was triggered even if `onRowClick` was undefined which meant it still changed the colour to show it as selected

File: packages/bpk-component-datatable/src/BpkDataTable.tsx [hunk 3]
```diff
@@ -15,20 +15,18 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-/* @flow strict */
 
-import PropTypes from 'prop-types';
 import type { KeyboardEvent } from 'react';
 import { useMemo, useState } from 'react';
+// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
```
Comment (anambl) [inline packages/bpk-component-datatable/src/BpkDataTable.tsx:21]: `react-table` is not typed. There are some community types, but they're not very accurate and are causing issues so went with `any` type whenever there's something from `react-table`

**Comment by anambl** [comment]
The Percy snapshots for mobile view changed slightly. That's because the `react-table` column has some default properties as well one of which is `minWidth` set to 0. The BpkDataTable is not at all optimised for mobile which is where this could be an issue (it was an issue before as well as even though there was no `minWidth`, the headers were no longer aligned with the columns). We could just set the `minWidth` to undefined instead, but I don't think that's an improvement so I'd leave it as is. The other default values from column are fine and don't impact the component

**Review by olliecurtis — APPROVED** [review]

**Pull request merged**